src/runtime/flow_node_handler/activity_handler/activity_handler.ts

```diff
@@ -75,27 +44,70 @@ export abstract class FlowNodeHandlerInterruptible<TFlowNode extends Model.Base.
 
     return new Promise<void>(async(resolve: Function, reject: Function): Promise<void> => {
       try {
+        this._previousFlowNodeInstanceId = previousFlowNodeInstanceId;
+        token.flowNodeInstanceId = this.flowNodeInstanceId;
+        let nextFlowNodes: Array<Model.Base.FlowNode>;
+
         this._terminationSubscription = this._subscribeToProcessTermination(token, reject);
         await this._attachBoundaryEvents(token, processTokenFacade, processModelFacade, identity, resolve);
 
-        await super.execute(token, processTokenFacade, processModelFacade, identity, previousFlowNodeInstanceId);
+        await this.beforeExecute(token, processTokenFacade, processModelFacade, identity);
+        nextFlowNodes = await this.executeInternally(token, processTokenFacade, processModelFacade, identity);
+        await this.afterExecute(token, processTokenFacade, processModelFacade, identity);
+
+        // EndEvents will return "undefined" as the next FlowNode.
+        // So if no FlowNode is to be run next, we have arrived at the end of the current flow.
+        const processIsNotYetFinished: boolean = nextFlowNodes && nextFlowNodes.length > 0;
```

Comment:
you could explicitly check for undefined

```suggestion
        const processIsNotYetFinished: boolean = nextFlowNodes !== undefined && nextFlowNodes.length > 0;
```

You could also move the undefined check to another const and remove the first comment line like this:

```suggestion
        const previousFlowNodeWasNoEndEvent: boolean = nextFlowNodes !== undefined;
        const processIsNotYetFinished: boolean = previousFlowNodeWasNoEndEvent && nextFlowNodes.length > 0;
```

Reply:
Yes to the first, but no to the second. Doesn't help readability at all, imho.